packages/firestore/test/unit/specs/perf_spec.test.ts

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Query } from '../../../src/core/query';
+import { doc, orderBy, path } from '../../util/helpers';
+
+import { describeSpec, specTest } from './describe_spec';
+import { spec } from './spec_builder';
+
+const STEP_COUNT = 10;
+
+describeSpec('Performance Tests:', ['benchmark'], () => {
+  specTest('Insert a new document', [], () => {
+    let steps = spec().withGCEnabled(false);
+    for (let i = 0; i < STEP_COUNT; ++i) {
+      steps = steps.userSets(`collection/{i}`, { doc: i }).writeAcks(i);
+    }
+    return steps;
+  });
+
+  specTest('Insert a new document and wait for snapshot', [], () => {
```

Comment:
More explicit: Start a listen, write a document, ack write, handle watch snapshot, unlisten [repeat ${STEP_COUNT} times]

Reply:
Updated this and most other test titles.